### ISSUE TITLE
[Bug]: No Data Loading for UK: midsussex_gov_uk

### ISSUE THREAD

**Issue by mark-summers** [mampfes/hacs_waste_collection_schedule#1454]
### I Have A Problem With:

A specific source

### What's Your Problem

No data loading for UK: Mid-Sussex District Council, looks like the website has changed and includes some Christmas dates so assume that's why it's not working, but the data is still there. Most likely a source defect. Logs attached.

![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/56111163/e10950a1-4505-44a3-9767-32975c294db3)

### Source (if relevant)

midsussex_gov_uk

### Logs

```Shell
Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136
Integration: waste_collection_schedule (documentation)
First occurred: 09:54:08 (1 occurrences)
Last logged: 09:54:08

fetch failed for source Mid-Sussex District Council: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/midsussex_gov_uk.py", line 99, in fetch date=datetime.strptime(td[1].text, "%A %d %B %Y").date(), ~~^^^ IndexError: list index out of range
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by djdobs** [comment]
Same here. Hasn’t been working for many days, assumed it was just me

**Comment by 5ila5** [comment]
fixed in #1463.
Fix is part of release 1.44.0

They do not update their collection table but offer a separate table showing changes. I now read in this table (if it exists) but it may not work next year if they change the layout or headline of the table. But reading the base table will probably not fail.

**Comment by mark-summers** [comment]
> fixed in #1463.
> 
> Fix is part of release 1.44.0
> 
> 
> 
> They do not update their collection table but offer a separate table showing changes. I now read in this table (if it exists) but it may not work next year if they change the layout or headline of the table. But reading the base table will probably not fail.
> 
> 

Thanks so much for your help!
